driver/task.go

```diff
@@ -55,6 +56,7 @@ type Task struct {
 	Version         string
 	UserDefinedMeta map[string]map[string]string
 	BufferPeriod    *BufferPeriod // nil when disabled
+	Condition       config.ConditionConfig
```

Comment:
No-op, but wanted to note my thoughts. When we start seeing this become more of a pattern to import the config package for using the underlying objects, we may need to look into a dedicated object package so the config package doesn't end up so pervasive in every core code paths.

Reply:
Agree! Thanks for mentioning this. 🤔 I spent awhile debating whether or not to reuse the config object here. On one hand it didn't seem worth it to build out a Condition interface with Services and CatalogServices implementations just to thread condition information through. On the other hand, to your point, importing config around isn't great. I think having a dedicated object package is a good idea that would solve this dilemma!

Reply:
Agreed on the trade offs you mentioned! We can continue keeping tabs on this to see how it plays out later